### PR TITLE
Add wrappers for the most relevant kraken features.

### PR DESCRIPTION
**Pull request by davebx**

### CODE REVIEW

File: tool_collections/kraken/kraken/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+How to test Kraken in your local galaxy instance:
+
+In your galaxy home directory config/tool_conf.xml file, change  
+
+```
+ <section name="Kraken-test" id="kraken">
```
Comment (bgruening) [inline tool_collections/kraken/kraken/README.md]: @davebx is this needed, an old left over? I have done this as well, but this is was way before TS :)

File: tool_collections/kraken/kraken/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+```
+<table name="krakendb" comment_char="#">
+    <columns>value ,dbkey, display_name, file_path</columns>
+    <file path="/full-path/Kraken_galaxy/tool-data/kraken_database.loc" />
+</table>
+```
+
+TODO: 
+
+1. Fix kraken.xml to take in fastq files
```
Comment (bgruening) [inline tool_collections/kraken/kraken/README.md]: It does take fastq files.

File: tool_collections/kraken/kraken_filter/kraken-filter.xml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<tool id="kraken-filter" name="Filter Kraken" version="1.0.0">
+    <description>
+        by confidence.
```
Comment (bgruening) [inline tool_collections/kraken/kraken_filter/kraken-filter.xml]: minus '.'

File: tool_collections/kraken/kraken_filter/kraken-filter.xml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        <import>macros.xml</import>
+    </macros>
+    <command>
+        <![CDATA[
+        kraken-filter @INPUT_DATABASE@ --threshold $threshold "${input}" > "$filtered_output"
+        ]]>
+    </command>
+    <inputs>
+        <param format="tabular" label="Kraken classified output" name="input" type="data" />
+        <param max="1" min="0" label="Confidence threshold" name="threshold" type="float" value="0" />
```
Comment (bgruening) [inline tool_collections/kraken/kraken_filter/kraken-filter.xml]: `float` -> min & value

File: tool_collections/kraken/kraken_report/.shed.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+owner: devteam
+name: kraken_report
+description: Contains galaxy tool wrapper for kraken-report.
+categories:
+  - Metagenomics
+long_description: 'Kraken is a system for assigning taxonomic labels to short DNA
```
Comment (bgruening) [inline tool_collections/kraken/kraken_report/.shed.yml]: Does this work in yaml? I thought we need to use `|` to indicate a multi-line text.

File: tool_collections/kraken/kraken_translate/kraken-translate.xml [hunk 6]
```diff
@@ -0,0 +1,44 @@
+<tool id="kraken-translate" name="Kraken-translate" version="1.0.0">
+    <description>
+        convert taxonomy IDs to names.
```
Comment (bgruening) [inline tool_collections/kraken/kraken_translate/kraken-translate.xml]: The dot can be removed.

File: tool_collections/kraken/kraken/README.md (hunk 2 above)
Comment (kosman1) [inline tool_collections/kraken/kraken/README.md]: Yes that's a dinosaur of a line, and you're welcome for it

**Comment by bgruening** [comment]
@davebx has the included histogram some really new features that the build in histogram is lacking? I would try as much as possible to separate plots from tools.

File: tool_collections/kraken/kraken_report/.shed.yml (hunk 5 above)
Comment (davebx) [inline tool_collections/kraken/kraken_report/.shed.yml]: It seems to work, planemo happily parses it when I run the shed_upload command.

**Comment by bgruening** [comment]
Besides the obvious missing tests, this looks good to me. I think we can merge this and create a Issue to not forget to write tests at some point.

**Comment by davebx** [comment]
I agree that the kraken tools should have tests, but unfortunately the tools depend on having a kraken database available, and the databases can take up a fair amount of space. I believe the bacteria library on test is around 150GB, which I'd say is a bit on the large side for a toolshed repository.

**Comment by bgruening** [comment]
@davebx I know. Had the same problem with diamond. But at some point we should have tests defined. Even a test for an error message "Database not found". Would at least check the wrapper logic.
If no one complains, I will merge this tomorrow.

**Pull request merged**